### ISSUE TITLE
Add full type definition for manifest types

### ISSUE THREAD

**Issue by aL3891** [Microsoft/service-fabric-client-dotnet#3]
In the old Sdk there the various manifest types are fully defined in terms of properties so that they can be deserialized directly from the xml manifests on disk.

It would be useful to have those in this library to be able to easily set things like versions and properties programmatically as part of a CI pipeline without dealing with xml namespaces and all that.

For example, compare System.Fabric.Management.ServiceModel.ApplicationManifestType to Microsoft.ServiceFabric.Common.ApplicationTypeManifest

**Comment by amanbha** [comment]
Thsi type represents the value returned by REST api call https://docs.microsoft.com/en-us/rest/api/servicefabric/sfclient-api-getapplicationmanifest#responses

**Comment by aL3891** [comment]
But the Service model dll doesn't have any dependencies of its own, its just a pure net standard library that happens to be bundled with the sdk, (I'm using a decompiled version of it in my projects in fact)
 
Having the static types for the manifests is very useful as I mentioned and there does not seem to be any dependency reason for why they could not be added to this library, or atleast be open sourced on their own

**Comment by amanbha** [comment]
Yes, there is no issue in using that, For this REST library, we don't want to add a dependency on Microsoft.ServiceFabric nuget package, but carrying the dll in this nuget package can be an option.

**Comment by aL3891** [comment]
I've started to spike this, but I don't have a PR yet, however one change i'd like to make is to rename ApplicationTypeManifest, in Microsoft.ServiceFabric.Common to something like TypeManifestResponse so that it doesn't collide with the xsd types. Also, what's your opinion on adding the xsd and a script to generate all the types?

**Comment by amanbha** [comment]
Let's not rename the types as most of this code is autogenerated from Service fabric REST swagger specification and not from xsd.
